firmware/controllers/system/efi_gpio.h

```diff
@@ -61,25 +62,31 @@ class OutputPin {
 
 
 #if EFI_GPIO_HARDWARE
-	ioportid_t port = 0;
-	uint8_t pin = 0;
 	#if (BOARD_EXT_GPIOCHIPS > 0)
 		/* used for external pins */
-		brain_pin_e brainPin;
 		bool ext;
 	#endif
 #endif /* EFI_GPIO_HARDWARE */
+	pin_output_mode_e outputMode;
+	brain_pin_e brainPin;
```

Comment:
field;
fieldPtr;

that's something really really unusual.

overall I am extremely confused by this PR. without adequate comments I see enormous amount of code complexity and risk, there is no way such complexity and risk justifies the benefits of this PR.

Reply:
same thing going on here as with ButtonDebounce...